mantisshrimp/models/mantis_rcnn/mantis_faster_rcnn.py

```diff
@@ -4,16 +4,54 @@
 from mantisshrimp.core import *
 from mantisshrimp.models.mantis_rcnn.rcnn_param_groups import *
 from mantisshrimp.models.mantis_rcnn.mantis_rcnn import *
+from mantisshrimp.backbones import *
 
 
 class MantisFasterRCNN(MantisRCNN):
     @delegates(FasterRCNN.__init__)
-    def __init__(self, n_class, h=256, pretrained=True, metrics=None, **kwargs):
+    def __init__(
+        self,
+        n_class,
+        h=256,
+        backbone="default",
+        pretrained=True,
+        metrics=None,
+        **kwargs,
+    ):
         super().__init__(metrics=metrics)
-        self.n_class, self.h, self.pretrained = n_class, h, pretrained
-        self.m = fasterrcnn_resnet50_fpn(pretrained=self.pretrained, **kwargs)
-        in_features = self.m.roi_heads.box_predictor.cls_score.in_features
-        self.m.roi_heads.box_predictor = FastRCNNPredictor(in_features, self.n_class)
+        self.n_class, self.h, self.backbone, self.pretrained = (
+            n_class,
+            h,
+            backbone,
+            pretrained,
+        )
+
+        if self.backbone == "defualt":
+            # Creates the default fasterrcnn as given in pytorch. Trained on COCO dataset
+            self.m = fasterrcnn_resnet50_fpn(pretrained=self.pretrained, **kwargs)
+            in_features = self.m.roi_heads.box_predictor.cls_score.in_features
+            self.m.roi_heads.box_predictor = FastRCNNPredictor(
+                in_features, self.n_class
+            )
+        else:
+            # Creates the custom backbone model trained on ImageNet.
+            try:
+                self.base_model = create_torchvision_backbone(
+                    backbone=self.backbone, is_pretrained=pretrained
+                )
+                self.ft_anchor_generator = AnchorGenerator(
```

Comment:
Where is this being used?

Reply:
Resolved from previous conversation. I am not putting them altogether allowing **kwargs.